matchzoo/engine/base_model.py

```diff
@@ -235,6 +235,8 @@ def _make_output_layer(self):
             return keras.layers.Dense(task.num_classes, activation='softmax')
         elif isinstance(task, tasks.Ranking):
             return keras.layers.Dense(1, activation='sigmoid')
```

Comment:
It's wonderful to bind the output with the task type. Here, I think the `sigmoid` is optimal in `tasks.Ranking`, since not all models require the output to be a probability. Sometimes, it is more efficient to train the model when the output to be a real number (e.g., in hinge loss).

Reply:
The point of moving layer constructions from tasks to models is to allow models to use different losses for the same task by overriding the default `_make_output_layer`. In your case where a model wants a different output activation function, the model will need to override `_make_output_layer` and handle `tasks.Ranking` output layer construction itself.

Reply:
Well, it's well enough in current `_make_output_layer` since it can cover the majority of ranking cases. Special requirement can be satisfied by overriding.

Reply:
@faneshion correct, that's the whole point of this PR.